src/python-kolibri-flatpak-utils/utils.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+
+import fcntl
+import io
+import os
+import socket
+import subprocess
+import urllib.request
+
+from contextlib import contextmanager
+from urllib.error import URLError
+
+
+XDG_DATA_HOME = os.environ.get('XDG_DATA_HOME', None)
+
+KOLIBRI_IDLE_TIMEOUT_MINS = int(os.environ.get("KOLIBRI_IDLE_TIMEOUT_MINS", 60))
+KOLIBRI_IDLE_TIMEOUT_SECS = KOLIBRI_IDLE_TIMEOUT_MINS * 60
+
+KOLIBRI_HTTP_PORT = int(os.environ.get("KOLIBRI_HTTP_PORT", 8080))
```

Comment:
Ah, okay, I'll use that! I feel a bit dirty about assuming `KOLIBRI_URL = "http://127.0.0.1:{}".format(KOLIBRI_HTTP_PORT)`, as well, but I figure that can be changed later if it's needed.

Reply:
Now I remember: the problem I had here is that kolibri.utils.conf has some code which tries to access (and then to create) KOLIBRI_HOME. That won't work in a socket-activated Kolibri setup since the desktop launcher doesn't have access to that part of the system. I added a bit of a workaround in the latest commit. Does it look okay to you?

Reply:
Ah, good catch. I added a comment on that commit with a suggested tweak: https://github.com/learningequality/org.learningequality.Kolibri/pull/2/commits/1c65eb23ebe493c3e94a00e1dc43f65db008f5ed